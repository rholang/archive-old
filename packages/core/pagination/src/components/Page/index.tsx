import React, { Component } from 'react';
import Button, { ButtonProps, ButtonAppearances } from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  ButtonProps,
  {
    appearance?: ButtonAppearances;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
