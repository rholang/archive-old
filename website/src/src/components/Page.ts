import styled from 'styled-components';
import { gridSize, math } from '@atlaskit/theme';

const PageContainer = styled.main`
  width: 100vw;
  max-width: calc(640px + 4rem);
  box-sizing: border-box;
  margin: 2rem auto;
  padding: 0 2rem;
`;

export default PageContainer;

export const Title = styled.h1`
  margin-bottom: 1em;
`;

export const Section = styled.section`
  margin-top: 3em;

  p {
    line-height: 1.4em;
  }
`;

export const Intro = styled.p`
  font-size: ${math.multiply(gridSize, 2)}px;
  font-weight: 300;
  line-height: 1.4em;
`;
