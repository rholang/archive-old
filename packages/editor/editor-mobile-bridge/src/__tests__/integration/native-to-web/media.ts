import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  callNativeBridge,
  editor,
  editable,
  getDocFromElement,
  skipBrowsers as skip,
} from '../_utils';

const fakeMobileId = `fake-aaaa-bbbb-cccc-dddddddddd`;

const uploadPreviewUpdatePayload = (dimensions?: {
  width: number;
  height: number;
}) => ({
  file: {
    dimensions,
    id: fakeMobileId,
    name: 'test-file.jpeg',
    type: 'image/jpeg',
  },
  preview: {
    dimensions,
  },
});

const mobileUploadEndPayload = (publicId: string, collectionName: string) => ({
  file: {
    id: fakeMobileId,
    publicId,
    collectionName,
    name: 'test-file.jpeg',
    type: 'image/jpeg',
  },
});

BrowserTestCase(
  `media.ts: Collection + Dimensions => uploading`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(
        uploadPreviewUpdatePayload({
          width: 2265,
          height: 1500,
        }),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Collection + Dimensions => complete`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(
        uploadPreviewUpdatePayload({
          width: 2265,
          height: 1500,
        }),
      ),
    );

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(
        mobileUploadEndPayload(
          '03907bf7-7dbb-408f-a334-27a5ae6bb7b9',
          'MediaServicesDemoCollection',
        ),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Collection + No dimension => uploading`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(uploadPreviewUpdatePayload()),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Collection + No dimension => complete`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(uploadPreviewUpdatePayload()),
    );

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(
        mobileUploadEndPayload(
          '03907bf7-7dbb-408f-a334-27a5ae6bb7b9',
          'MediaServicesDemoCollection',
        ),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Empty collection + Dimensions => uploading`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(
        uploadPreviewUpdatePayload({
          width: 2265,
          height: 1500,
        }),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Empty collection + Dimensions => complete`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(
        uploadPreviewUpdatePayload({
          width: 2265,
          height: 1500,
        }),
      ),
    );

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(
        mobileUploadEndPayload('03907bf7-7dbb-408f-a334-27a5ae6bb7b9', ''),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Empty collection + No dimension => uploading`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(uploadPreviewUpdatePayload()),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Empty collection + No dimension => complete`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(
        uploadPreviewUpdatePayload({
          width: 2265,
          height: 1500,
        }),
      ),
    );

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(
        mobileUploadEndPayload('03907bf7-7dbb-408f-a334-27a5ae6bb7b9', ''),
      ),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
