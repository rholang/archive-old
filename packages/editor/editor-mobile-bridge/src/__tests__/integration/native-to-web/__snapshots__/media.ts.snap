// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media.ts: Collection + Dimensions => complete 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "MediaServicesDemoCollection",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Collection + Dimensions => uploading 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "InitialCollectionForTesting",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Collection + No dimension => complete 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "MediaServicesDemoCollection",
            "height": 200,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 250,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Collection + No dimension => uploading 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "InitialCollectionForTesting",
            "height": 200,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 250,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Empty collection + Dimensions => complete 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Empty collection + Dimensions => uploading 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "InitialCollectionForTesting",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Empty collection + No dimension => complete 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Empty collection + No dimension => uploading 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "InitialCollectionForTesting",
            "height": 200,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 250,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
